homeassistant/components/upcloud/__init__.py

```diff
@@ -271,18 +271,13 @@ def icon(self) -> str:
         """Return the icon of this server."""
         return "mdi:server" if self.is_on else "mdi:server-off"
 
-    @property
-    def state(self) -> str | None:
-        """Return state of the server."""
-        try:
-            return STATE_MAP.get(self._server.state, self._server.state)
-        except AttributeError:
-            return None
-
     @property
     def is_on(self) -> bool:
         """Return true if the server is on."""
-        return self.state == STATE_ON
+        try:
+            return STATE_MAP.get(self._server.state, self._server.state) == STATE_ON
+        except AttributeError:
+            return False
```

Comment:
Shouldn't this be checked when the state is polled, setting the entity unavailable?

Reply:
Maybe, I made the change as-is, as in, I don't know much about this thing either.
I only move current existing logic.

Reply:
Yeah, but before this change this would result in STATE_UNKNOWN, instead of STATE_OFF?
Not sure if it matters though.

Reply:
Oddly enough, binary sensors & switches don't have an unknown state.

Reply:
Pushed a change to extend the availability property to handle that.